### ISSUE TITLE
Agent and Server mode share flag pointers

### ISSUE THREAD

**Issue by roidelapluie** [prometheus/prometheus#9648]
```
Default("data/").StringVar(&cfg.localStoragePath)
Default("data-agent/").StringVar(&cfg.localStoragePath)
```
They should point to different variables. i actually do not even know what happens when I launch prometheus without options.
other agent flags should be checked too.

**Comment by roidelapluie** [comment]
cc @rfratto for awareness

**Comment by roidelapluie** [comment]
closed by #9660
